src/control/cmd/dmg/support.go

```diff
@@ -0,0 +1,182 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/cmd/dmg/pretty"
+	"github.com/daos-stack/daos/src/control/lib/control"
+	"github.com/daos-stack/daos/src/control/lib/support"
+)
+
+// NetCmd is the struct representing the top-level network subcommand.
+type SupportCmd struct {
+	CollectLog collectLogCmd `command:"collectlog" description:"Collect logs from servers"`
+}
+
+// collectLogCmd is the struct representing the command to collect the logs from the servers for support purpose
+type collectLogCmd struct {
+	baseCmd
+	cfgCmd
+	ctlInvokerCmd
+	hostListCmd
+	jsonOutputCmd
+	Stop         bool   `short:"s" long:"stop" description:"Stop the collectlog command on very first error"`
+	TargetFolder string `short:"t" long:"target" description:"Target Folder location where log will be copied"`
+	Archive      bool   `short:"z" long:"archive" description:"Archive the log/config files"`
+	CustomLogs   string `short:"c" long:"custom-logs" description:"Collect the Logs from given directory"`
+}
+
+func (cmd *collectLogCmd) Execute(_ []string) error {
+	// Default log collection set
+	var LogCollection = map[string][]string{
+		"CopyServerConfig":     {""},
+		"CollectSystemCmd":     support.SystemCmd,
+		"CollectServerLog":     support.ServerLog,
+		"CollectDaosServerCmd": support.DaosServerCmd,
+	}
+
+	// Default 7 set of support collection steps to show in progress bar
+	progress := support.ProgressBar{1, 7, 0, cmd.jsonOutputEnabled()}
```

Comment:
Not a fan of the magic numbers again. This time I'm not sure where the 7 comes from.